qiskit/transpiler/passes/synthesis/high_level_synthesis.py

```diff
@@ -349,12 +350,12 @@ def _run(
 
             elif node.op.name == "reset":
                 # reset qubits to 0
-                tracker.set_clean(context.to_globals(qubits))
+                tracker.set_clean(global_qubits)
                 processed = True
 
             # check if synthesis for the operation can be skipped
-            elif self._definitely_skip_node(node, qubits, dag):
-                tracker.set_dirty(context.to_globals(qubits))
+            elif self._definitely_skip_node(node, global_qubits, dag):
+                tracker.set_dirty(global_qubits)
 
             # next check control flow
             elif node.is_control_flow():
```

Comment:
@alexanderivrii: I can't comment there, but in the line below we have
```python
inner_context = context.restrict(qubits)
```
and here it failed if I used the `global_qubits`. Does this make sense to you?